AccessControl: Document basic roles changes and provisioning V2 (#48910)
grafana/grafana: docs/sources/enterprise/access-control/assign-rbac-roles.md

</br>

**To assign a role to a team:**

1. Open the YAML configuration file.

1. Refer to the following table to add attributes and values.

>    <!-- TODO find out what LINK was meant to go there -->

Reviewer: No link is meant to go here. The step is telling the user to refer to (or use) the information in the table to add attributes and values.

Author: I was referring to this line, that was previously in the docs with a `[LINK]` placeholder:
```md
Enter the permissions `action` and `scope` values. For more information about permissions actions and scopes, refer to [LINK]
```

Author: Figured it out!

   | `roles > permissions`    | Enter the permissions `action` and `scope` values. For more information about permissions actions and scopes, refer to [LINK] |
   | `teams`                  | Enter the team or teams to which you are adding the custom role.                                                              |
   | `teams > orgId`          | Because teams belong to organizations, you must add the `orgId` value.                                                        |
   | `teams > name`           | Enter the name of the team.                                                                                                   |
   | `teams > roles`          | Enter the custom or fixed role or roles that you want to grant to the team.                                                   |
   | `teams > roles > name`   | Enter the name of the role.                                                                                                   |
   | `teams > roles > global` | Enter `true`, or specify `orgId` of the role you want to assign to the team. Fixed roles are global.                          |

>    For more information about managing custom roles refer to [Create custom roles using provisioning]({{< relref "./manage-rbac-roles/#create-custom-roles-using-provisioning" >}}).

Reviewer: ```suggestion
   For more information about managing custom roles, refer to [Create custom roles using provisioning]({{< relref "./manage-rbac-roles/#create-custom-roles-using-provisioning" >}}).
```

        scope: 'users:*'
      - action: 'org.users:write'
        scope: 'users:*'

# Assignments to teams
teams:
  - name: 'user writers'
    orgId: 1
    roles:
      # Custom role assignment
      - name: 'custom:users:writer'
        global: true
      # Fixed role assignment
      - name: 'fixed:users:writer'
        global: true
  - name: 'user admins'
    orgId: 1
    roles:
      - name: 'custom:users:writer'
        global: true
      - name: 'fixed:users:writer'
        global: true
```

> **Note**: The roles don't have to be defined in the provisioning configuration files to be assigned. As long as they exist in the database, they are assignable.

Reviewer: ```suggestion
> **Note**: The roles don't have to be defined in the provisioning configuration files to be assigned. If roles exist in the database, they can be assigned.
```

**Remove a role assignment from a team:**

If you want to remove an assignment from a team, add `state: absent` to the `teams > roles` section, and reload the configuration file.

> The following example creates the `custom:users:writer` role and assigns it to the `user admins` team along with the `fixed:users:writer` role, but ensures the `user writers` team does not have them:

Reviewer: I'm not sure what "but ensures the user writers team does not have them" means. Doesn't have what?

Author: Maybe not the best phrasing I agree :thinking: 
Provisioning will remove `custom:users:writer` and `fixed:users:writer` assignments to `user writers` if such assignments exist.

Reviewer: Gotcha, ok - I think that sentence is trying to achieve quite a bit. Maybe break it up, something like:

The following example:
-  Creates the `custom:users:writer` role
- Assigns the `custom:users:writer` role and the `fixed:users:writer` role to the `user admin` team
- Removes the `custom:users:writer` and the `fixed:users:writer` assignments from the `user writers` team, if those assignments exist.

```yaml
# config file version
apiVersion: 2

# Roles to insert/update in the database
roles:
  - name: 'custom:users:writer'
    description: 'List/update other users in the organization'
    version: 1
    global: true
    permissions:
      - action: 'org.users:read'
        scope: 'users:*'
      - action: 'org.users:write'
        scope: 'users:*'

# Assignments to teams
teams:
  - name: 'user writers'
    orgId: 1
    roles:
      - name: 'fixed:users:writer'
        global: true
        state: 'absent' # Remove assignment
      - name: 'custom:users:writer'
        global: true
        state: 'absent' # Remove assignment
  - name: 'user admins'
    orgId: 1
    roles:
      - name: 'fixed:users:writer'
        global: true
      - name: 'custom:users:writer'
        global: true
```

> **Note**: The roles don't have to be defined in the provisioning configuration files to be revoked. As long as they exist in the database, they are revokable.

Reviewer: ```suggestion
> **Note**: The roles don't have to be defined in the provisioning configuration files to be revoked. If roles exist in the database, they can be revoked.
```
